### PR TITLE
Address some warnings

### PR DESCRIPTION
**Pull request by sksamuel**

### CODE REVIEW

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/future/FutureMatcherTest.kt [hunk 1]
```diff
@@ -2,17 +2,24 @@ package com.sksamuel.kotest.matchers.future
...
 import kotlinx.coroutines.withContext
 import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Executors
 
 class FutureMatcherTest : StringSpec({
+
    suspend fun runOnSeparateThread(block: () -> Unit) {
-      @OptIn(DelicateCoroutinesApi::class)
-      newSingleThreadContext("separate").use {
+      Executors.newSingleThreadExecutor().asCoroutineDispatcher().use {
```
Comment (OliverO2) [inline kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/future/FutureMatcherTest.kt]: Why are we going back to explicit JDK executors?

File: kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/test/interceptors/CoroutineDispatcherInterceptorTest.kt [hunk 2]
```diff
@@ -32,9 +32,8 @@ class CoroutineDispatcherInterceptorTest : DescribeSpec() {
             )
 
             val controller = object : CoroutineDispatcherFactory {
-               @OptIn(DelicateCoroutinesApi::class)
                override suspend fun <T> withDispatcher(testCase: TestCase, f: suspend () -> T): T =
-                  newSingleThreadContext("foo").use { dispatcher ->
+                  Executors.newSingleThreadExecutor().asCoroutineDispatcher().use { dispatcher ->
```
Comment (OliverO2) [inline kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/test/interceptors/CoroutineDispatcherInterceptorTest.kt]: Why are we going back to explicit JDK executors?

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/future/FutureMatcherTest.kt (hunk 1 above)
Comment (sksamuel) [reply kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/future/FutureMatcherTest.kt]: I just figured newSingleThreadContext has the warnings, the jvm code doesn't, and I thought they were the same basically. I'll switch it back, but curious if the kotlin one has advantages?

File: kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/test/interceptors/CoroutineDispatcherInterceptorTest.kt (hunk 2 above)
Comment (sksamuel) [reply kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/test/interceptors/CoroutineDispatcherInterceptorTest.kt]: Reverted

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/future/FutureMatcherTest.kt (hunk 1 above)
Comment (sksamuel) [reply kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/future/FutureMatcherTest.kt]: Reverted

Comment (OliverO2) [reply kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/future/FutureMatcherTest.kt]: > curious if the kotlin one has advantages?

I usually do some research when selecting such stuff and then quickly forget all the details. In this case, I'd say the dispatchers provided by kotlinx.coroutines
* are multiplatform,
* create named threads on the JVM (easier debugging),
* use a slightly different executor on the JVM (and my assumption is that the coroutines folks usually have sound reasons for their choices).

**Review by OliverO2 — APPROVED** [review]

**Pull request merged**